src/pages/nft/claim/index.vue

```diff
@@ -1424,38 +1433,51 @@ export default {
       }
     },
     async ensureClassIdInCollected() {
-      this.isViewCollectionLoading = true;
-
-      let collectedNFTs = this.getCollectedNFTClassesByAddress(
-        this.claimingAddress
-      );
+      if (this.classId && this.isAutoDeliver) {
+        this.isViewCollectionLoading = true;
+        let collectedNFTs = this.getCollectedNFTClassesByAddress(
+          this.claimingAddress
+        );
+        let ownCount =
+          this.getNFTClassOwnerInfoById(this.classId)[this.claimingAddress]
+            ?.length || 0;
 
-      if (!collectedNFTs?.some(item => item.classId === this.classId)) {
-        for (let attempts = 0; attempts < MAX_ATTEMPT; attempts += 1) {
-          try {
-            // eslint-disable-next-line no-await-in-loop
-            await this.fetchCollectedNFTClassesByAddress({
-              address: this.claimingAddress,
-              nocache: true,
-            });
+        if (!collectedNFTs?.some(item => item.classId === this.classId)) {
+          for (let attempts = 0; attempts < MAX_ATTEMPT; attempts += 1) {
+            try {
+              // eslint-disable-next-line no-await-in-loop
+              await Promise.all([
+                this.fetchCollectedNFTClassesByAddress({
+                  address: this.claimingAddress,
+                  nocache: true,
+                }),
+                this.fetchNFTOwners({ classId: this.classId, nocache: true }),
+              ]);
```

Comment:
I think we can push conditionally

Reply:
[🥅 Ensure owners are fetched if necessary](https://github.com/likecoin/liker-land/pull/2057/commits/119ed204096e1c2ae68b391b918176edc6571b5e)